pkg/pgmodel/extension.go

```diff
@@ -234,11 +256,20 @@ func getSqlVersion(v semver.Version, extName string) string {
 }
 
 // getNewExtensionVersion returns the highest version allowed by validRange
-func getNewExtensionVersion(extName string, availableVersions semver.Versions, validRange semver.Range, validCurrentVersion bool, currentVersion semver.Version) (semver.Version, bool) {
+func getNewExtensionVersion(extName string,
+	availableVersions semver.Versions,
+	defaultVersion semver.Version,
+	validRange semver.Range,
+	validCurrentVersion bool,
+	currentVersion semver.Version) (semver.Version, bool) {
 	//sort higher extensions first
 	sort.Sort(sort.Reverse(availableVersions))
 	printedWarning := false
 	for i := range availableVersions {
+		/* skip any versions above the default version for auto-upgrade */
+		if availableVersions[i].GT(defaultVersion) {
```

Comment:
Could you please mention the significance of the `defaultVersion` here? The name default does not self explain the matter to me.

Reply:
Default version is specified by the extension itself when installed into the system. It is usually the latest version but not always. It is installed by CREATE EXTENSION if no version is explicitly specified.